tests/linen/linen_test.py

```diff
@@ -442,20 +442,20 @@ def __call__(self, x):
     {
       'model_index': 2,
       'key_paths': {
-        'MultiHeadDotProductAttention_0/key/bias/u',
-        'MultiHeadDotProductAttention_0/key/kernel/u',
-        'MultiHeadDotProductAttention_0/out/kernel/u',
-        'MultiHeadDotProductAttention_0/query/bias/u',
-        'MultiHeadDotProductAttention_0/query/kernel/u',
-        'MultiHeadDotProductAttention_0/value/bias/u',
-        'MultiHeadDotProductAttention_0/value/kernel/u',
-        'MultiHeadDotProductAttention_0/key/bias/sigma',
-        'MultiHeadDotProductAttention_0/key/kernel/sigma',
-        'MultiHeadDotProductAttention_0/out/kernel/sigma',
-        'MultiHeadDotProductAttention_0/query/bias/sigma',
-        'MultiHeadDotProductAttention_0/query/kernel/sigma',
-        'MultiHeadDotProductAttention_0/value/bias/sigma',
-        'MultiHeadDotProductAttention_0/value/kernel/sigma',
+        'MultiHeadAttention_0/key/bias/u',
```

Comment:
a bit worried that we might break a lot of checkpoints

Reply:
This could also break code that indexes into params, since the class name is changed:
```
>>> class A(nn.Module):
...   @nn.compact
...   def __call__(self, x):
...     x = nn.MultiHeadAttention(2)(x)
...     return x
... 
>>> x = jnp.ones((2, 3, 4))
>>> a = A()
>>> v = a.init(jax.random.key(0), x)
>>> v
{'params': {'MultiHeadAttention_0': ...}} # instead of MultiHeadDotProductAttention
```
To prevent breakage, we could force the name to still be `MultiHeadDotProductAttention`, but that may not be intuitive to users since it doesn't actually align with the class name. We could also keep `MultiHeadDotProductAttention` and just add `MultiHeadAttention` as an alias (rather than the other way around), but it could still be confusing for users using `MultiHeadAttention`.